### PR TITLE
docs script

### PR DESCRIPTION
**Pull request by ktoso**

### CODE REVIEW

File: scripts/generate_docs.sh [hunk 1]
```diff
@@ -16,21 +15,24 @@
...
-modules=(CoreMetrics Metrics)
+declare -r modules=(CoreBaggage)
+main_module="CoreBaggage"
+declare -r project_xcodeproj_name="swift-distributed-tracing-baggage-core"
+declare -r project_gh_name="swift-distributed-tracing-baggage-core"
 
 if [[ "$(uname -s)" == "Linux" ]]; then
   # build code if required
-  if [[ ! -d "$root_path/.build/x86_64-unknown-linux" ]]; then
+  if [[ ! -d "$root_path/.build/x86_64-unknown-linux-gnu" ]]; then
```
Comment (ktoso) [inline scripts/generate_docs.sh:27]: this is when running ` docker-compose -f docker/docker-compose.yaml -f docker/docker-compose.1804.53.yaml run shell` it seems ubuntu is now doing the `-gnu`

File: scripts/generate_docs.sh [hunk 2]
```diff
@@ -16,21 +15,24 @@
...
 
 my_path="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
-root_path="$my_path/.."
+declare -r root_path="$my_path/.."
 version=$(git describe --abbrev=0 --tags || echo "0.0.0")
-modules=(CoreMetrics Metrics)
+declare -r modules=(CoreBaggage)
+main_module="CoreBaggage"
+declare -r project_xcodeproj_name="swift-distributed-tracing-baggage-core"
+declare -r project_gh_name="swift-distributed-tracing-baggage-core"
```
Comment (ktoso) [inline scripts/generate_docs.sh:23]: centralized values a bit

File: scripts/generate_docs.sh [hunk 3]
```diff
@@ -16,21 +15,24 @@
...
 
 if [[ "$(uname -s)" == "Linux" ]]; then
   # build code if required
-  if [[ ! -d "$root_path/.build/x86_64-unknown-linux" ]]; then
+  if [[ ! -d "$root_path/.build/x86_64-unknown-linux-gnu" ]]; then
     swift build
   fi
   # setup source-kitten if required
-  source_kitten_source_path="$root_path/.SourceKitten"
+  source_kitten_source_path="$root_path/.build/.SourceKitten"
```
Comment (ktoso) [inline scripts/generate_docs.sh:31]: was polluting the project dir

File: scripts/generate_docs.sh [hunk 4]
```diff
@@ -39,13 +41,13 @@ if [[ "$(uname -s)" == "Linux" ]]; then
   mkdir -p "$root_path/.build/sourcekitten"
   for module in "${modules[@]}"; do
     if [[ ! -f "$root_path/.build/sourcekitten/$module.json" ]]; then
-      "$source_kitten_path/sourcekitten" doc --spm-module $module > "$root_path/.build/sourcekitten/$module.json"
+      "$source_kitten_path/sourcekitten" doc --spm --module-name $module > "$root_path/.build/sourcekitten/$module.json"
```
Comment (ktoso) [inline scripts/generate_docs.sh:44]: sourcekitten changed their command line interface...

File: scripts/generate_docs.sh [hunk 5]
```diff
@@ -88,11 +81,17 @@ for module in "${modules[@]}"; do
...
   if [[ -f "$root_path/.build/sourcekitten/$module.json" ]]; then
     args+=(--sourcekitten-sourcefile "$root_path/.build/sourcekitten/$module.json")
   fi
+
   jazzy "${args[@]}"
+
+  if [[ $(cat $root_path/.build/sourcekitten/$module.json) == "0" ]]; then
+    echo "processing module $module ($root_path/.build/sourcekitten/$module.json) yielded EMPTY module json file!"
+    exit 1
+  fi
```
Comment (ktoso) [inline scripts/generate_docs.sh:94]: added an eager failure if we end up generating zero symbols

**Comment by ktoso** [comment]
These are the fixes I had to do to make it work again @tomerd https://github.com/apple/swift-distributed-tracing-baggage-core/pull/4/commits/d3795ec4cf121e7e60918b33d99bd2e638267435

We should replicate the sourcekitten (or really even just this form of script) to other projects IMO.

**Review by slashmo — APPROVED** [review]
LGTM 👍

**Pull request merged**